### PR TITLE
Add KeepAlive logic that auto-reconnects

### PR DESCRIPTION
**Pull request by egordon**
Test procedure:

1. Run F/T Sensor
2. Run the program: `ros2 run forque_sensor_hardware forque_sensor_hardware`
3. Verify successful initalization
4. Turn off the F/T sensor, verify that this prints the warning "KeepAlive Failed, re-init"
5. Turn on the F/T sensor, verify that, eventually, Initialization is successful (it may take a couple tries as the F/T sensor boots up.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -235,31 +257,51 @@ class WirelessFTNode : public rclcpp::Node
...
+  }
+
   // For reading from F/T Sensor and publishing wrench
-  void timer_callback()
-  {
+  void timer_callback() {
     auto packet = mWFT->readDataPacket();
     if (!packet.valid) {
-      RCLCPP_WARN(get_logger(), "Skipping Invalid Packet");
+      //RCLCPP_WARN(get_logger(), "Skipping Invalid Packet");
```
Comment (amalnanavati) [inline src/main.cpp:278]: Can you keep this in, at least as a debug log? I've found it valuable to indicate that there may be an error in the FT sensor.

**Review by amalnanavati — CHANGES_REQUESTED** [review]
This is an interesting architecture. As far as I can tell, every 0.5 secs, you set `mIsAlive` to false, and then every timer cycle, if you successfully read from telnet, you set it to true.

My main concern with the code as-is is that I've seen errors where it can successfully read from telnet (it gets F/T data) but has trouble writing to telnet (setting bias). That case doesn't seem to be captured by this code yet. 

I'd want to test this in-person over a meal to see how often it reconnects, and how much it delays the F/T rate, to know whether/how to adjust other system components accordingly.

But this does seem like a strictly beneficial change and I think we should use it for our study tonight!

**Comment by egordon** [comment]
I don't think I can non-ambiguously figure out a way to solve your case where UDP F/T packets are still coming in but TCP telnet commands don't work. So we can talk about it later.

Do we want to re-init the TCP connection only to try and keep the UDP connection alive? That might work if we do it entirely within the service call, though if the UDP also dies while re-connecting telnet it might get tricky. Do we re-enable everything and intentionally temporarily lose UDP connectivity in order to re-enable telnet? That is easier and safer, but then we almost certainly will trip the watchdog.

Plus this will all be a nightmare to test as the bug is hard to consistently re-create.

**Comment by egordon** [comment]
TODO: it took a while to re-connect. (Maybe because of router issues)

**Review by amalnanavati — APPROVED** [review]

**Pull request merged**